child_compassion/wizards/global_child_search.py

```diff
@@ -69,20 +69,16 @@ def do_search(self):
         # Remove previous search results
         self.global_child_ids = False
 
-        mapping = base_mapping.new_onramp_mapping(self._name, self.env)
+        mapping = base_mapping.new_onramp_mapping(self._name, self.env,
+                                                  "profile_search")
         params = mapping.get_connect_data(self)
         onramp = OnrampConnector()
         result = onramp.send_message(
             'beneficiaries/availabilitysearch', 'GET', None, params)
         if result['code'] == 200:
             self.nb_found = result['content']['NumberOfBeneficiaries']
-            children = result['content']['BeneficiarySearchResponseList']
-            mapping = base_mapping.new_onramp_mapping(
-                'compassion.global.child', self.env)
-            for child_data in children:
-                child_vals = mapping.get_vals_from_connect(child_data)
-                self.global_child_ids += self.env[
-                    'compassion.global.child'].create(child_vals)
+            self.map_and_create_json_oject(result,
+                                           'BeneficiarySearchResponseList')
```

Comment:
I'm going in details but why not directly pass `result['content']['BeneficiarySearchResponseList']` to the function ?